### PR TITLE
deprecate the class "Algebra"

### PR DESCRIPTION
**Pull request by fchapoton**
deprecate the auld-style "Algebra" class, and replace its last use in sage codebase by `Parent`

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.

### CODE REVIEW

**Comment by mkoeppe** [comment]
Docbuild failure is unrelated (#38467)

File: src/sage/rings/ring.pyx [hunk 1]
```diff
@@ -86,6 +86,17 @@ This is to test a deprecation::
...
     sage: F.category()
     Category of principal ideal domains
+
+    sage: from sage.rings.ring import Algebra
+    sage: class Nichts(Algebra):
+    ....:     pass
+    sage: F = Nichts(QQ, QQ)
+    ...:
+    DeprecationWarning: use the category Algebras
+    See https://github.com/sagemath/sage/issues/38000 for details.
```
Comment (mkoeppe) [inline src/sage/rings/ring.pyx]:
```suggestion
    See https://github.com/sagemath/sage/issues/38502 for details.
```

File: src/sage/rings/ring.pyx [hunk 2]
```diff
@@ -1474,24 +1485,11 @@ cdef class Field(CommutativeRing):
...
-            <sage.rings.ring.Algebra object at ...>
-        """
-        # This is a low-level class. For performance, we trust that the category
-        # is fine, if it is provided. If it isn't, we use the category of Algebras(base_ring).
-        if category is None:
-            category = check_default_category(Algebras(base_ring), category)
-        Ring.__init__(self,base_ring, names=names, normalize=normalize,
-                      category=category)
+    def __init__(self, base_ring, *args, **kwds):
+        deprecation(38000, "use the category Algebras")
```
Comment (mkoeppe) [inline src/sage/rings/ring.pyx]:
```suggestion
        deprecation(38502, "use the category Algebras")
```

File: src/sage/rings/ring.pyx [hunk 3]
```diff
@@ -1474,24 +1485,11 @@ cdef class Field(CommutativeRing):
...
 cdef class Algebra(Ring):
-    """
-    Generic algebra
-    """
-    def __init__(self, base_ring, names=None, normalize=True, category=None):
-        """
-        Initialize ``self``.
-
-        EXAMPLES::
+    _default_category = _Rings
```
Comment (mkoeppe) [inline src/sage/rings/ring.pyx]: Does this not break user code, as there's nothing that sets the correct category?

**Comment by fchapoton** [comment]
now triggers an unexpected failure in `src/sage/categories/category_with_axiom.py`...

File: src/sage/rings/ring.pyx [hunk 4]
```diff
@@ -70,12 +70,12 @@ This is to test a deprecation::
     sage: from sage.rings.ring import CommutativeAlgebra
     sage: class Nein(CommutativeAlgebra):
     ....:     pass
-    sage: F = Nein(QQ, QQ)
+    sage: F = Nein(QQ)
```
Comment (mkoeppe) [inline src/sage/rings/ring.pyx:73]: why this change?

File: src/sage/categories/category_with_axiom.py [hunk 5]
```diff
@@ -1908,7 +1908,7 @@ def _base_category_class_and_axiom(cls):
             sage: CommutativeRings()._base_category_class_and_axiom
             (<class 'sage.categories.rings.Rings'>, 'Commutative')
             sage: CommutativeRings()._base_category_class_and_axiom_origin
-            'deduced by base_category_class_and_axiom'
+            'set by __classget__'
```
Comment (mkoeppe) [inline src/sage/categories/category_with_axiom.py:1911]: do you understand how this change comes about?

File: src/sage/rings/ring.pyx (hunk 4 above)
Comment (fchapoton) [reply src/sage/rings/ring.pyx:73]: to enforce similarity with the Algebra case ; the only mandatory argument is the basering

File: src/sage/categories/category_with_axiom.py (hunk 5 above)
Comment (fchapoton) [reply src/sage/categories/category_with_axiom.py:1911]: no. It may come from no longer using this category in CommutativeAlgebra,  possibly.

File: src/sage/rings/ring.pyx (hunk 3 above)
Comment (fchapoton) [reply src/sage/rings/ring.pyx]: better now

File: src/sage/rings/ring.pyx [hunk 6]
```diff
@@ -1474,32 +1487,18 @@ cdef class Field(CommutativeRing):
 
 
 cdef class Algebra(Ring):
-    """
-    Generic algebra
-    """
-    def __init__(self, base_ring, names=None, normalize=True, category=None):
-        """
-        Initialize ``self``.
```
Comment (mkoeppe) [inline src/sage/rings/ring.pyx:1482]: Why remove the docstring?

Comment (fchapoton) [reply src/sage/rings/ring.pyx:1482]: To make it very clear that this is *really deprecated and essentially removed*.
